lighthouse-core/audits/dobetterweb/doctype.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+
+class Doctype extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      name: 'doctype',
+      description: 'Page has the HTML doctype',
+      failureDescription: 'Page is missing the HTML doctype',
+      helpText: 'Specifying a doctype prevents the browser from switching to quirks-mode.' +
+                'Read more on the ' +
+                '[MDN Web Docs page](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)',
+      requiredArtifacts: ['Doctype'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    if (artifacts.Doctype === null) {
+      return {
+        rawValue: false,
+      };
+    }
+
+    if (artifacts.Doctype.name.trim().length === 0) {
+      return {
+        rawValue: false,
+        explanation: 'Doctype name cannot be empty',
+      };
+    }
+
+    /* Note that the value for name is case sensitive,
+       and must be the string `html`. For details see:
+       https://html.spec.whatwg.org/multipage/parsing.html#the-initial-insertion-mode */
+    if (artifacts.Doctype.name.trim() === 'html') {
+      return {
+        rawValue: true,
+      };
+    } else {
+      return {
+        rawValue: false,
+        explanation: 'Doctype name must be the lowercase string html',
+      };
+    }
+  }
```

Comment:
I think @paulirish also wanted to give audit feedback on the "`publicId` and `systemId` should be empty string cases" to get full coverage of the spec? Though is that only xhtml or how does that even happen :)

Reply:
@paulirish You want me to add checks for those additional properties in this audit? Thanks!

Reply:
yup!

this applies to HTML pages as well. just make assert those params are empty string.